akka-http-core/src/main/scala/akka/http/impl/engine/client/pool/NewHostConnectionPool.scala

```diff
@@ -349,7 +360,13 @@ private[client] object NewHostConnectionPool {
 
           def dispatchResponseResult(req: RequestContext, result: Try[HttpResponse]): Unit = logic.dispatchResponseResult(req, result)
 
-          def willCloseAfter(res: HttpResponse): Boolean = logic.willClose(res)
+          def willCloseAfter(res: HttpResponse): Boolean = {
+            logic.willClose(res) || keepAliveTimeApplies()
```

Comment:
This will make the `slot` assume the connection will be closed after this response has been delivered - but will it also actually make sure the connection is closed?

Reply:
After testing it manually it worked like, it would close, but now i thinking about it. I might used a too low idle limit, which means _that_ closed it actually. I will get back to this after doing some manual testing with the new parameter and `idle-timeout`

Reply:
With `idle-timeout` set to `infinite` it is still works

Reply:
I tested with this server: https://gist.github.com/zsedem/7fbab5e62335c907aeafcda3da04de47